include/benchmark/benchmark_api.h

```diff
@@ -429,6 +543,35 @@ class State {
   BENCHMARK_ALWAYS_INLINE
   size_t iterations() const { return total_iterations_; }
 
+
+  // Add/set a user counter. Return the counter handle.
+  BENCHMARK_ALWAYS_INLINE
+  size_t SetCounter(Counter const& c) { return counters_.Set(c); }
+  // Add/set a user counter. Return the counter handle.
+  BENCHMARK_ALWAYS_INLINE
+  size_t SetCounter(const char* n, double v = 0., uint32_t f = Counter::kDefaults) { return counters_.Set(n, v, f); }
+  // Add/set a user counter. Return the counter handle.
+  BENCHMARK_ALWAYS_INLINE
+  size_t SetCounter(std::string const& n, double v = 0., uint32_t f = Counter::kDefaults) { return counters_.Set(n, v, f); }
+  // Given a counter handle, set a previously existing user counter.
+  BENCHMARK_ALWAYS_INLINE
+  void   SetCounter(size_t counter_id, double v) { counters_.Get(counter_id).Set(v); }
```

Comment:
consider reducing this API surface to the essential. If we want users to define a Counter type, then don't supply overrides to do it for them. it just adds complexity that's unnecessary.